chainstate/test-suite/src/tests/homomorphism.rs

```diff
@@ -0,0 +1,187 @@
+// verifier per operation and then combining the result.
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn coins_homomorphism(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let mut rng = make_seedable_rng(seed);
+
+        let storage1 = Store::new_empty().unwrap();
+        let mut tf = TestFramework::builder()
+            .with_storage(storage1.clone())
+            .with_tx_verification_strategy(TxVerificationStrategy::Default)
+            .build();
+
+        let storage2 = Store::new_empty().unwrap();
+        let mut tf2 = TestFramework::builder()
+            .with_storage(storage2.clone())
+            .with_tx_verification_strategy(TxVerificationStrategy::Disposable)
+            .build();
+
+        let tx_1 = TransactionBuilder::new()
+            .add_input(
+                TxInput::new(
+                    OutPointSourceId::BlockReward(tf.genesis().get_id().into()),
+                    0,
+                ),
+                empty_witness(&mut rng),
+            )
+            .add_output(TxOutput::new(
+                OutputValue::Coin(Amount::from_atoms(rng.gen_range(100_000..200_000))),
+                OutputPurpose::Transfer(anyonecanspend_address()),
+            ))
+            .build();
+
+        let tx_2 = TransactionBuilder::new()
+            .add_input(
+                TxInput::new(
+                    OutPointSourceId::Transaction(tx_1.transaction().get_id()),
+                    0,
+                ),
+                InputWitness::NoSignature(None),
+            )
+            .add_output(TxOutput::new(
+                OutputValue::Coin(Amount::from_atoms(rng.gen_range(1000..2000))),
+                OutputPurpose::Transfer(anyonecanspend_address()),
+            ))
+            .build();
+
+        let tx_3 = TransactionBuilder::new()
+            .add_input(
+                TxInput::new(
+                    OutPointSourceId::Transaction(tx_2.transaction().get_id()),
+                    0,
+                ),
+                InputWitness::NoSignature(None),
+            )
+            .add_output(TxOutput::new(
+                OutputValue::Coin(Amount::from_atoms(rng.gen_range(100..200))),
+                OutputPurpose::Transfer(anyonecanspend_address()),
+            ))
+            .build();
+
+        tf.make_block_builder()
+            .add_transaction(tx_1.clone())
+            .add_transaction(tx_2.clone())
+            .add_transaction(tx_3.clone())
+            .build_and_process()
+            .unwrap()
+            .unwrap();
+
+        tf2.make_block_builder()
+            .add_transaction(tx_1)
+            .add_transaction(tx_2)
+            .add_transaction(tx_3)
+            .build_and_process()
+            .unwrap()
+            .unwrap();
+
+        assert_eq!(storage1.dump_raw(), storage2.dump_raw());
+    });
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn tokens_homomorphism(#[case] seed: Seed) {
```

Comment:
I wonder whether we need more variations... at least throw in TODOs for that if you don't want to do that right now. I think all output types must go in here in a way, and I'm thinking this needs more thought on whether this should be done manually or in some generic form that guarantees that all new types are covered.

Reply:
I'm not talking about tokens, specifically, btw.

Reply:
Perhaps a connection strategy that does the merges randomly solves this problem. The one I mentioned before in my comment in disposable_strategy